src/backend/opencl/jit/BufferNode.hpp

```diff
@@ -9,14 +9,27 @@
 
 #pragma once
 #include <common/jit/BufferNodeBase.hpp>
-#include <common/jit/Node.hpp>
-#include <af/defines.h>
-#include <iomanip>
-#include <mutex>
 #include "../kernel/KParam.hpp"
 
+#include <memory>
+
 namespace opencl {
 namespace jit {
 using BufferNode = common::BufferNodeBase<std::shared_ptr<cl::Buffer>, KParam>;
 }
 }  // namespace opencl
+
+namespace common {
+
+template<typename DataType, typename ParamType>
+bool BufferNodeBase<DataType, ParamType>::operator==(
+    const BufferNodeBase<DataType, ParamType> &other) const noexcept {
+    if (m_data.get() == other.m_data.get()) {
+        if (m_bytes == other.m_bytes) {
+            if (m_param.offset == other.m_param.offset) { return true; }
+        }
+    }
+    return false;
```

Comment:
```suggestion
    return m_data.get() == other.m_data.get() &&
           m_bytes == other.m_bytes &&
           m_param.offset == other.m_param.offset;
```
Perhaps both generate same binary, but isn't above version much easier to read ?